### PR TITLE
KONFLUX-12087 [TaskRun Details] display results value correctly

### PR DESCRIPTION
**Pull request by rrosatti**
## Fixes 

Fixes https://issues.redhat.com/browse/KONFLUX-12087

## Description

The bug reported in the ticket was happening because UI was not expecting the `result.value` field from TaskRun's `results` array to be different than a string. That's the error it was vomiting:

```bash
Objects are not valid as a React child (found: object with keys {deps-i686, build-i686, deps-s390x, build-s390x, deps-x86_64, build-x86_64, deps-aarch64, deps-ppc64le, build-aarch64, build-ppc64le}). If you meant to render a collection of children, use an array instead.

```

I was not able to reproduce this issue with any TaskRuns in my tenant. So, I forced it to happen (see more info in `"How to test or reproduce?"` section somehwere below).

This is the TaskRun link reported in the ticket: https://konflux-ui.apps.kflux-rhel-p01.fc38.p1.openshiftapps.com/ns/rhel-pt-c-libs-tenant/applications/rhel-10/taskruns/rhel-build-ef366f496b8accd68f7d5a06392bad2a4d58-process-sources

Notice the TaskRun has the following results:

```json
    "results": [
      {
        "name": "skip-mpc-tasks",
        "type": "object",
        "value": {
          "deps-i686": "localhost",
          "build-i686": "localhost",
          "deps-s390x": "linux/s390x",
          "build-s390x": "linux/s390x",
          "deps-x86_64": "linux/amd64",
          "build-x86_64": "linux/amd64",
          "deps-aarch64": "linux/arm64",
          "deps-ppc64le": "linux/ppc64le",
          "build-aarch64": "linux/arm64",
          "build-ppc64le": "linux/ppc64le"
        }
      },
      {
        "name": "dependencies-artifact",
        "type": "string",
        "value": "oci:quay.io/redhat-user-workloads/rhel-pt-c-libs-tenant/rhel-10-flex@sha256:7da0116593617cfb3861d7325572f0432e86174b9e3cbf4d7b5b2acecaaaa634"
      }
    ],
```

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Before:

taskrun-details-results-ISSUE

After:

taskrun-details-results-FIX

## How to test or reproduce?

- you need to have a TaskRun with a object as one of its results' value
- if you don't find any, you can force it with the diff below:

```diff
diff --git a/src/components/TaskRunDetailsView/tabs/TaskRunDetailsTab.tsx b/src/components/TaskRunDetailsView/tabs/TaskRunDetailsTab.tsx
index f0acdc98..96c6a6d3 100644
--- a/src/components/TaskRunDetailsView/tabs/TaskRunDetailsTab.tsx
+++ b/src/components/TaskRunDetailsView/tabs/TaskRunDetailsTab.tsx
@@ -56,7 +56,26 @@ const TaskRunDetailsTab: React.FC = () => {
   }

   const taskRunFailed = (getTRLogSnippet(taskRun) || {}) as ErrorDetailsWithStaticLog;
-  const results = isTaskV1Beta1(taskRun) ? taskRun?.status?.taskResults : taskRun?.status?.results;
+  const r = isTaskV1Beta1(taskRun) ? taskRun?.status?.taskResults : taskRun?.status?.results;
+  const results = [
+    ...r,
+    {
+      name: 'skip-mpc-tasks',
+      type: 'object',
+      value: {
+        'deps-i686': 'localhost',
+        'build-i686': 'localhost',
+        'deps-s390x': 'linux/s390x',
+        'build-s390x': 'linux/s390x',
+        'deps-x86_64': 'linux/amd64',
+        'build-x86_64': 'linux/amd64',
+        'deps-aarch64': 'linux/arm64',
+        'deps-ppc64le': 'linux/ppc64le',
+        'build-aarch64': 'linux/arm64',
+        'build-ppc64le': 'linux/ppc64le',
+      },
+    },
+  ];
   const hasStart = typeof taskRun?.status?.startTime === 'string' && !!taskRun.status.startTime;
   const hasEnd =
     typeof taskRun?.status?.completionTime === 'string' && !!taskRun.status.completionTime;
```

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

## Summary by CodeRabbit

* **New Features**
  * Added a value-normalization utility for consistent display of parameter and result values.
* **Improvements**
  * Standardized formatting for arrays, objects, null/undefined, and primitive result/parameter values in the UI.
* **Tests**
  * Added comprehensive tests covering rendering, normalization, failure states, and edge-case value handling.

### CODE REVIEW

**Review by testcara — APPROVED** [review]
LGTM, tests passed locally.
